Automattic/wp-calypso
# A4A: Refactor SitesDataViews, extract code to decouple it from A4A and Jetpack

Resolves: https://github.com/Automattic/automattic-for-agencies-dev/issues/312

## Proposed Changes

This PR decouples the Sites Dashboard components to make it neutral and shareable across Calypso. Dotcom wants to use it in its site redesign.

In this PR, we are decoupling the `SitesDataViews,` the most coupled component to Jetpack and A4A elements. We also needed to decouple A4A because each feature family has to configure its own columns and actions to be rendered by the `DataViews` component.

**Components Changes Overview:**

- `SitesDashboard`: Updated to use the new `JetpackSitesDataViews`
- `JetpackSitesDataViews`: New component to configure the columns/actions for the `ItemsDataView` for the A4A/Jetpack context
- `ItemsDataViews`: New common component to be shared across Calypso (i.e. Dotcom)

## Testing Instructions

- The main test is to check that everything is working as before. There are not any behavior or visual change
- Check the code to familiarize yourself with it.

## Pre-merge Checklist
